src/amount.rs

```diff
@@ -87,63 +88,100 @@ pub type AmountCounter = u32;
 ///
 #[derive(Clone, Debug, Copy, Default, Serialize, Deserialize)]
 pub struct Amount {
-    pub count: AmountCounter,
-    pub unit: PowerOfTen,
+    count: AmountCounter,
+    unit: PowerOfTen,
 }
 
-/// A NormalizedAmount is just like an Amount except that count is a Big Integer.
-/// So sum or difference of any two Amounts sharing the same unit can be represented
-/// with a NormalizedAmount.
+/// A NormalizedAmount is just like an Amount except that count is a u128
+/// So sum or difference of any two compatible Amounts sharing the same unit can
+/// be represented with a NormalizedAmount.
 ///
 /// For now, have this only for internal use/ops.
-#[derive(Debug)]
+#[derive(Debug, Default)]
 struct NormalizedAmount {
-    count: Integer,
+    count: u128,
     unit: PowerOfTen,
 }
 
 impl Amount {
-    pub fn new(count: AmountCounter, unit: PowerOfTen) -> Self {
-        // We constrain count to ::counter_max().  If you want to use a bigger value,
-        // you must change the unit.
-        debug_assert!(count <= Self::counter_max());
+    /// power-of-ten exponent representing COUNT_MAX
+    /// max value for Self::count is 10^9
+    //  nine digits per tx ought to be enough for anybody! -- danda 2021.
+    const COUNT_MAX_TEN_POW: PowerOfTen = 9;
+
+    /// creates a new Amount.
+    ///   count: count of 10^unit values.
+    ///   unit:  power-of-ten exponent, as used in 10^unit
+    ///
+    /// Returns Error::AmountInvalid if count > Self::counter_max()
+    #[inline]
+    pub fn new(count: AmountCounter, unit: PowerOfTen) -> Result<Self> {
+        if count > Self::counter_max() {
+            return Err(Error::AmountInvalid);
+        }
+
+        Ok(Self { count, unit })
+    }
 
+    // only for use in this crate because code in this impl depends on
+    // count always being <= Self::counter_max().
+    pub(crate) fn new_unchecked(count: AmountCounter, unit: PowerOfTen) -> Self {
+        debug_assert!(count <= Self::counter_max());
         Self { count, unit }
     }
 
+    // returns count of 10^unit
+    #[inline]
```

Comment:
You've added `#[inline]` to a lot of these functions, I'm curious if you noticed an impact on performance?